e2e/builder/build_test.go

```diff
@@ -130,11 +131,34 @@ func TestKitMaxBuildLimit(t *testing.T) {
 
 				// verify that all builds are successful
 				Eventually(BuildPhase(ns, buildA), TestTimeoutLong).Should(Equal(v1.BuildPhaseSucceeded))
+				Eventually(BuildConditions(ns, buildA), TestTimeoutLong).ShouldNot(BeNil())
```

Comment:
I don't think it's really required a check on this condition, as it will eventually get scheduled. What may be more interesting is to check that a build which is supposed to wait, have this condition turned on, and later move to scheduled.